WordPress/gutenberg
# [RN Mobile] RichText: Return early when content has changed in onSelectionChangeFromAztec

## What?
If the component's content has changed, return early in the `RichText::onSelectionChangedFromAztec` method.

## Why?

There have been reports of slow performance in the RichText component on older and lower-end devices. One potential cause of the slow performance is the amount of work that happens with each keypress in the `RichText::onSelectionChangeFromAztec` method.

Issue: https://github.com/wordpress-mobile/gutenberg-mobile/issues/2594

## How?

It seems unnecessary to do anything in the `RichText::onSelectionChangeFromAztec` method while the user is currently typing quickly. It appears that the `RichText:onChangeFromAztec` method is only called at the beginning and end of a typing spree. Therefore, a potential improvement is to only update the selection when `RichText::onChangeFromAztec` has been called. We can determine whether `RichText::onChangeFromAztec` has been called by comparing the event's content to the component's current content and checking the last event type. If they are different and the last event type was 'selection change', `RichText::onChangeFromAztec` was not called. (If it had been called, it would have set the component's content to the event's content [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/rich-text/src/component/index.native.js#L341)). 

I expect this change to significantly decrease the lag experienced when creating a new block after typing. I also expect this change to decrease the lag experienced when a block using the RichText component expands to create a new line after a significant amount of typing.

## Testing Instructions
The [writing flows](https://github.com/wordpress-mobile/test-cases/tree/trunk/test-cases/gutenberg/writing-flow) will need to be tested thoroughly to verify that this change doesn't cause any regressions.

## Screenshots or screencast 
n/a
